pkg/controller/collection/collection_controller.go

```diff
@@ -465,6 +467,24 @@ func getNamespaceForObject(u *unstructured.Unstructured, defaultNamespace string
 
 func reconcileActiveVersions(collectionResource *kabanerov1alpha1.Collection, collections []resolvedCollection, c client.Client) error {
 
+	// Gather the known collection asset (*-tasks, *-pipeline) substitution data.
+	renderingContext := make(map[string]interface{})
+	if len(collections) > 0 {
+		// The collection id is the name of the Appsody stack directory ("the stack name from the stack path").
+		// Appsody stack creation namimg constraints:
+		// "The name must start with a lowercase letter, contain only lowercase letters, numbers, or dashes,
+		// and cannot end in a dash."
+		cID := collections[0].collection.Id
+		if !cIDRegex.MatchString(cID) {
+			return fmt.Errorf("collection Activation failed because an invalid collection id of %v was found. The collection id value must follow stack creation name rules", cID)
```

Comment:
is there someplace or some thing that we can point the user to, to show what the rules are?  I don't know if Appsody has a web page or something we can link to.